cyberdrop_dl/crawlers/ashemaletube.py

```diff
@@ -0,0 +1,170 @@
+VIDEO_SELECTOR = "video > source"
+PROFILE_VIDEOS_SELECTOR = "div.media-item__inner a[data-video-preview]"
+NEXT_PAGE_SELECTOR = "a.rightKey"
+MODEL_VIDEO_SELECTOR = "a data-video-preview"
+USER_NAME_SELECTOR = "h1.username"
+PLAYLIST_VIDEOS_SELECTOR = "a.playlist-video-item__thumbnail"
+DATETIME_SELECTOR = "div.views-count-add"
+JS_SELECTOR = "script:contains('var player = new VideoPlayer')"
+RESOLUTIONS = ["2160p", "1440p", "1080p", "720p", "480p", "360p", "240p"]  # best to worst
+INCLUDE_VIDEO_ID_IN_FILENAME = True
+
+
+class Format(NamedTuple):
+    resolution: str
+    link_str: str
+
+
+class AShemaleTubeCrawler(Crawler):
+    primary_base_domain = PRIMARY_BASE_DOMAIN
+
+    def __init__(self, manager: Manager) -> None:
+        super().__init__(manager, "ashemaletube", "aShemaletube")
+
+    """~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"""
+
+    @create_task_id
+    async def fetch(self, scrape_item: ScrapeItem) -> None:
+        """Determines where to send the scrape item based on the url."""
+        if "videos" in scrape_item.url.parts:
+            return await self.video(scrape_item)
+        elif "model" in scrape_item.url.parts:
+            return await self.model(scrape_item)
+        elif "playlists" in scrape_item.url.parts:
+            return await self.playlist(scrape_item)
+        elif "profiles" in scrape_item.url.parts:
+            return await self.profile(scrape_item)
+        raise ValueError
+
+    @error_handling_wrapper
+    async def playlist(self, scrape_item: ScrapeItem) -> None:
+        async for soup in self.web_pager(scrape_item):
+            if playlist_name := soup.select_one("h1"):
+                play_list_name_str = playlist_name.get_text().replace("\n", "").replace("\t", "").strip()
+                title = f"{play_list_name_str} [playlist]"
+                title = self.create_title(title)
+                scrape_item.setup_as_album(title)
+            await self.iter_videos(scrape_item, soup.select(PLAYLIST_VIDEOS_SELECTOR))
+
+    @error_handling_wrapper
+    async def profile(self, scrape_item: ScrapeItem) -> None:
+        return await self.model(scrape_item)
+
+    @error_handling_wrapper
+    async def model(self, scrape_item: ScrapeItem) -> None:
+        async for soup in self.web_pager(scrape_item):
+            if model_name := soup.select_one(USER_NAME_SELECTOR):
+                title = model_name.get_text().strip()
+                title = self.create_title(title)
+                scrape_item.setup_as_profile(title)
+            await self.iter_videos(scrape_item, soup.select(PROFILE_VIDEOS_SELECTOR))
+
+    async def iter_videos(self, scrape_item: ScrapeItem, videos) -> None:
+        for video in videos:
+            link: URL = create_canonical_video_url(video.get("href"))
+            new_scrape_item = scrape_item.create_child(link)
+            await self.video(new_scrape_item)
```

Comment:
The `video` method makes a new request. Send it to the queue instead of awaiting it.

```suggestion
            self.manager.task_group.create_task(self.run(new_scrape_item))
```